std/haxe/Template.hx

```diff
@@ -65,6 +65,9 @@ class Template {
 	**/
 	public static var globals : Dynamic = {};
 
+	// To avoid issues with DCE, keep the array iterator.
+	@:keep var hxKeepArrayIterator = [].iterator();
```

Comment:
I think the proper way to keep array iterator for `haxe.Template` is to add `@:ifFeature(haxe.Template)` to the `Array.iterator` field.

Reply:
In which file should that be added?

Reply:
I'm not sure if that should be added to the `std/Array.hx` or to each target Array implementation.
Maybe someone knows a better way. I just feel that the proposed behavior to create two useless objects per template instance is even worse.
Let's wait for the third opinion :)

Reply:
I can make it `static` if that helps. But if `@:ifFeature` is the way to go, Ill (try to) change to that (when I know how) 👍

Reply:
You have a valid concern regarding Array-Template relations. I guess the "opposite" @:ifFeature meta should exist. Maybe @simn knows.